src/tables/colr.rs

```diff
@@ -47,16 +185,523 @@ impl FromData for LayerRecord {
+    pub x1: f32,
+    /// The `y1` value of the radial gradient.
+    pub y1: f32,
+    /// The extend radial gradient.
+    pub extend: GradientExtend,
+    #[cfg(feature = "variable-fonts")]
+    variation_data: VariationData<'a>,
+    color_line: ColorLine<'a>,
+}
+
+impl<'a> core::fmt::Debug for RadialGradient<'a> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("RadialGradient")
+            .field("x0", &self.x0)
+            .field("y0", &self.y0)
+            .field("r0", &self.r0)
+            .field("r1", &self.r1)
+            .field("x1", &self.x1)
+            .field("y1", &self.y1)
+            .field("extend", &self.extend)
+            // TODO: Avoid hardcoding foregrounf color here?
+            // .field("stops", &self.stops(0))
+            .finish()
+    }
+}
+
+impl<'a> RadialGradient<'a> {
+    /// Returns an iterator over the stops of the radial gradient.
+    pub fn stops<'b>(
+        &'b self,
+        palette: u16,
+        #[cfg(feature = "variable-fonts")] coords: &'a [NormalizedCoordinate],
+    ) -> GradientStopsIter<'a, 'b> {
+        GradientStopsIter {
+            color_line: &self.color_line,
+            palette,
+            index: 0,
+            #[cfg(feature = "variable-fonts")]
+            variation_data: self.variation_data,
+            #[cfg(feature = "variable-fonts")]
+            coords,
+        }
+    }
+}
+
+/// A [sweep gradient](https://learn.microsoft.com/en-us/typography/opentype/spec/colr#formats-8-and-9-paintsweepgradient-paintvarsweepgradient)
+#[derive(Clone)]
+pub struct SweepGradient<'a> {
+    /// The x of the center of the sweep gradient.
+    pub center_x: f32,
+    /// The y of the center of the sweep gradient.
+    pub center_y: f32,
+    /// The start angle of the sweep gradient.
+    pub start_angle: f32,
+    /// The end angle of the sweep gradient.
+    pub end_angle: f32,
+    /// The extend of the sweep gradient.
+    pub extend: GradientExtend,
+    #[cfg(feature = "variable-fonts")]
+    variation_data: VariationData<'a>,
+    color_line: ColorLine<'a>,
+}
+
+impl<'a> core::fmt::Debug for SweepGradient<'a> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("SweepGradient")
+            .field("center_x", &self.center_x)
+            .field("center_y", &self.center_y)
+            .field("start_angle", &self.start_angle)
+            .field("end_angle", &self.end_angle)
+            .field("extend", &self.extend)
+            // TODO: Avoid hardcoding foregrounf color here?
+            // .field("stops", &self.stops(0))
+            .finish()
+    }
+}
+
+impl<'a> SweepGradient<'a> {
+    /// An iterator over the stops of the sweep gradient.
+    // TODO: Make API nicer (same for radial and linear gradient)
+    pub fn stops<'b>(
+        &'b self,
+        palette: u16,
+        #[cfg(feature = "variable-fonts")] coords: &'a [NormalizedCoordinate],
```

Comment:
It's not so nice that you have to manually pass the coords for this, instead of them being taken directly from the face. i tried it without it but it turned into a bit of a lifetime mess. So I will leave this as a potential improvement for the future.

Reply:
Sure. No problem.